spec/controllers/users/omniauth_callbacks_controller_spec.rb

```diff
@@ -8,20 +8,21 @@
 
   before do
     OmniAuth.config.test_mode = true
-    OmniAuth.config.mock_auth[:psu] = oauth_response
+    OmniAuth.config.mock_auth[:azure_oauth] = oauth_response
 
     request.env['devise.mapping'] = Devise.mappings[:user]
-    request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:psu]
+    request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:azure_oauth]
 
     allow(User).to receive(:from_omniauth)
       .with(oauth_response)
       .and_return(user)
   end
 
-  describe '#psu' do
+  describe '#azure_oauth' do
     context 'when the user persists correctly' do
       before do
-        get :psu
+        allow(user).to receive(:persisted?).and_return(true)
```

Comment:
This seems a little suspect. If we're trying to test if a user persists correctly, this could be bypassing that.